test/jdk/java/lang/reflect/code/TestPrimitiveTypePatterns.java

```diff
@@ -0,0 +1,88 @@
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.code.*;
+import java.lang.reflect.code.bytecode.BytecodeGenerator;
+import java.lang.reflect.code.interpreter.Interpreter;
+import java.lang.reflect.code.op.ExtendedOp;
+import java.lang.reflect.code.type.JavaType;
+import java.lang.reflect.code.type.PrimitiveType;
+import java.util.List;
+
+import static java.lang.reflect.code.op.CoreOp.*;
+import static java.lang.reflect.code.op.ExtendedOp.*;
+import static java.lang.reflect.code.type.FunctionType.*;
+
+/*
+ * @test
+ * @run testng TestPrimitiveTypePatterns
+ * @enablePreview
+ */
+
+public class TestPrimitiveTypePatterns {
+
+    @DataProvider
+    public static Object[][] patternsOfInt() {
+        return new Object[][]{
+                {JavaType.BYTE, new int[] {Byte.MIN_VALUE, Byte.MAX_VALUE, Byte.MIN_VALUE -1, Byte.MAX_VALUE + 1}},
+                {JavaType.SHORT, new int[] {Short.MIN_VALUE, Short.MAX_VALUE, Short.MIN_VALUE -1, Short.MAX_VALUE + 1}},
+                {JavaType.CHAR, new int[] {Character.MIN_VALUE, Character.MAX_VALUE, Character.MIN_VALUE -1, Character.MAX_VALUE + 1}},
+        };
+    }
+
+    @Test(dataProvider = "patternsOfInt")
+    void testPatternsOfInt(JavaType targetType, int[] values) throws Throwable {
+
+        var model = buildTypePatternModel(JavaType.INT, targetType);
+        var lmodel = model.transform(OpTransformer.LOWERING_TRANSFORMER);
+
+        var mh = BytecodeGenerator.generate(MethodHandles.lookup(), lmodel);
+
+        for (int v : values) {
+            Assert.assertEquals(Interpreter.invoke(lmodel, v), mh.invoke(v));
+        }
+    }
+
+    static FuncOp buildTypePatternModel(JavaType sourceType, JavaType targetType) {
+        // builds the model of:
+        // static boolean f(sourceType a) { return a instanceof targetType _; }
+        return func("f", functionType(JavaType.BOOLEAN, sourceType)).body(fblock -> {
```

Comment:
A different approach to explicitly building the model is to reflect on code in Java method and transform it. The compiler already knows how to build models. Might be a good test case for transforming. Perhaps consider as a follow up PR?

Reply:
Seems a good approach, I will implement it.